library/src/lib/time/time.component.ts

```diff
@@ -51,56 +52,103 @@ export class TimeComponent implements OnChanges, ControlValueAccessor {
 
     /** @hidden */
     @Output()
-    focusArrowLeft: EventEmitter<any> = new EventEmitter<any>();
+    readonly focusArrowLeft: EventEmitter<void> = new EventEmitter<void>();
 
-    /** @hidden */
+    /** @hidden
+     * Used only in meridian mode, stores information if it's AM or PM
+     * */
     period: string;
 
+    /** @hidden
+     * Defines if actually written meridian is invalid
+     * */
+    periodInvalid: boolean;
+
+    /** @hidden
+     * Variable that is displayed as an hour.
+     * For meridian mode ranging from 0 to 12,
+     * For non-meridian mode ranging from 0 to 23, and reflects the hour value
+     * */
+    displayedHour: number;
+
     /** @hidden */
-    oldPeriod: string;
+    onChange = (time: TimeObject) => {
+    };
 
     /** @hidden */
-    periodInvalid: boolean;
+    onTouched = () => {
+    };
 
     /** @hidden */
-    displayedHour: number;
+    registerOnChange(fn: (time: TimeObject) => void): void {
+        this.onChange = fn;
+    }
 
     /** @hidden */
-    @Input()
-    setDisplayedHour() {
-        if (this.time.hour === 0) {
+    registerOnTouched(fn: any): void {
+        this.onTouched = fn;
+    }
+
+    /** @hidden */
+    setDisabledState(isDisabled: boolean): void {
+        this.disabled = isDisabled;
+    }
+
+    constructor(
+        public timeI18nLabels: TimeI18nLabels,
+        public timeI18n: TimeI18n
+    ) {}
+
+    /** @hidden */
+    writeValue(time: TimeObject): void {
+        if (!time) {
+            return;
+        }
+        this.time = time;
+        this.setDisplayedHour();
+    }
+
+    /** @hidden */
+    ngOnChanges(): void {
```

Comment:
check for a specific type of changes? Or does this need to occur after every kind of change?

Reply:
Right now it checks if changes are including meridian/time configuration,